gtsam/hybrid/HybridBayesNet.cpp

```diff
@@ -164,9 +76,10 @@ DecisionTreeFactor HybridBayesNet::pruneDiscreteConditionals(
 }
 
 /* ************************************************************************* */
-HybridBayesNet HybridBayesNet::prune(size_t maxNrLeaves) {
+HybridBayesNet HybridBayesNet::prune(size_t maxNrLeaves) const {
+  HybridBayesNet copy(*this);
   DecisionTreeFactor prunedDiscreteProbs =
-      this->pruneDiscreteConditionals(maxNrLeaves);
+      copy.pruneDiscreteConditionals(maxNrLeaves);
```

Comment:
Why a copy if everything is functional?

Reply:
You're right, it was left-over ! I'll fix

Reply:
Actually, I messed up the discrete part. Did you ever profile the pruning as it is? It's an exponential operation - that alone might be responsible for slowness in some experiments.

I'll fix but we have to chat about this more I think.

Reply:
I haven't. I would like to see your perf-profiling setup and make it standard across projects. :) Seems to be very useful.

Reply:
OK, I fixed it. 

@varunagrawal :
- You probably had pruning tests in your project, but I added some here so I could refactor HGC::prune based on ADT::max. 
- You might want to check out his branch and try whether your experiments still work before we merge?